controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -772,6 +773,10 @@ protected void executeSingleTargetChild(PathAddress base, PathElement currentEle
             if (resource.hasChild(currentElement) || (nr != null && nr.isRemote())) {
                 safeExecute(next, newRemaining, context, nr, ignoreMissing);
             }
+            //if we are on the wrong host no need to do anything
+            if(!resource.hasChild(currentElement) && !HOST.equals(currentElement.getKey()) && (newRemaining != null && newRemaining.size() > 0)) {
+               throw new Resource.NoSuchResourceException(currentElement);
+            }
```

Comment:
I can't reproduce the issue any more. If I remember correctly this was happening during address resolution in multitarget calls